src/lib.rs

```diff
@@ -136,6 +193,45 @@ impl Escrow {
 
         reply(EscrowEvent::Cancelled(wallet_id));
     }
+
+    /// Continues cached transaction by `transaction_id`.
+    ///
+    /// Execution makes sense if, when returning from an async message,
+    /// the gas ran out and the state has changed.
+    async fn continue_transaction(&mut self, transaction_id: u64) {
+        let transactions = self.transactions.clone();
+        let payload = &transactions
+            .get(&transaction_id)
+            .expect("Transaction does not exist");
+        if let Some(action) = payload {
+            match action {
+                EscrowAction::Deposit(wallet_id) => {
+                    self.deposit(Some(transaction_id), *wallet_id).await
+                }
+                EscrowAction::Confirm(wallet_id) => {
+                    self.confirm(Some(transaction_id), *wallet_id).await
+                }
+                EscrowAction::Refund(wallet_id) => {
+                    self.refund(Some(transaction_id), *wallet_id).await
+                }
+                _ => unreachable!(),
+            }
+        } else {
+            msg::reply(EscrowEvent::TransactionProcessed, 0)
+                .expect("Error in a reply `EscrowEvent::TransactionProcessed`");
+        }
+    }
+
+    fn get_transaction_id(&mut self, transaction_id: Option<u64>) -> u64 {
+        match transaction_id {
+            Some(transaction_id) => transaction_id,
+            None => {
+                let transaction_id = self.transaction_id;
+                self.transaction_id = self.transaction_id.saturating_add(1);
```

Comment:
I think `wrapping_add` will be better because with IDs `saturating_add` will run out someday
```suggestion
                self.transaction_id = self.transaction_id.wrapping_add(1);
```